Rock/Reporting/DataSelect/GroupMember/GroupMemberAttributeSelect.cs

```diff
@@ -0,0 +1,419 @@
+                get
+                {
+                    if ( string.IsNullOrWhiteSpace( AttributeKey ) )
+                    {
+                        return false;
+                    }
+
+                    return true;
+                }
+            }
+
+            protected override void OnSetParameters( int version, IReadOnlyList<string> parameters )
+            {
+                // Parameter 1: Attribute Key
+                AttributeKey = DataComponentSettingsHelper.GetParameterOrEmpty( parameters, 0 );
+            }
+
+            protected override IEnumerable<string> OnGetParameters()
+            {
+                var settings = new List<string>();
+
+                settings.Add( AttributeKey.ToStringSafe() );
+
+                return settings;
+            }
+        }
+
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// Gets the entity type that the filter applies to.
+        /// </summary>
+        /// <value>
+        /// The namespace-qualified Type name of the entity that the filter applies to, or an empty string if the filter applies to all entities.
+        /// </value>
+        public override string AppliesToEntityType
+        {
+            get { return typeof( Model.GroupMember ).FullName; }
+        }
+
+        /// <summary>
+        /// Gets the name of the section in which the filter should be displayed in a browsable list.
+        /// </summary>
+        /// <value>
+        /// The section name.
+        /// </value>
+        public override string Section
+        {
+            get { return "Common"; }
+        }
+
+        /// <summary>
+        /// The PropertyName of the property in the anonymous class returned by the SelectExpression
+        /// </summary>
+        /// <value>
+        /// The name of the column property.
+        /// </value>
+        public override string ColumnPropertyName
+        {
+            get { return "Group Member Attribute"; }
+        }
+
+        #endregion
+
+        #region Public Methods
+
+        /// <summary>
+        /// Gets the title of the DataSelectComponent.
+        /// Override this property to specify a Title that is different from the ColumnPropertyName.
+        /// </summary>
+        /// <param name="entityType"></param>
+        /// <returns></returns>
+        /// <value>
+        /// The title.
+        /// </value>
+        public override string GetTitle( Type entityType )
+        {
+            return "Group Member Attribute";
+        }
+
+        private const string _CtlGroup = "pnlGroupAttributeFilterControls";
+        private const string _CtlProperty = "ddlProperty";
+
+        /// <summary>
+        /// Creates the child controls.
+        /// </summary>
+        /// <param name="parentControl"></param>
+        /// <returns></returns>
+        public override Control[] CreateChildControls( Control parentControl )
+        {
+            var pnlGroupAttributeFilterControls = new Panel();
```

Comment:
@mikepetersonccv Nothing so complicated, just ignorance on my part! I copied this from some other Rock code, wasn't aware of the alternatives. So I should use DynamicControlsPanel instead of Panel for pnlGroupAttributeFilterControls?